app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -362,23 +364,30 @@ function toRowField(name: string, valueType?: ValueType | null | undefined) {
   }
 }
 
-function getAstPattern(selector: string | number, action?: string) {
+function getAstPattern(selector: string | number, action?: string, castValueTypes?: string) {
+  const castSelector =
+    castValueTypes === 'number' && !isNaN(Number(selector)) ? Number(selector) : selector
```

Comment:
This could be done at the call site to keep the `getAstPattern` interface simpler.